dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -931,13 +936,18 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
         WorkflowVersion workflowVersion = workflowVersionDAO.findById(workflowVersionId);
         if (workflowVersion == null) {
             LOG.error(user.getUsername() + ": could not find version: " + workflow.getPath());
-            throw new CustomWebApplicationException("Version not found.", HttpStatus.SC_BAD_REQUEST);
+            throw new CustomWebApplicationException("Version not found.", HttpStatus.SC_INTERNAL_SERVER_ERROR);
 
         }
 
         List<Token> tokens = checkOnZenodoToken(user);
         Token zenodoToken = Token.extractToken(tokens, TokenType.ZENODO_ORG);
-        final String zenodoAccessToken = zenodoToken == null ? null : zenodoToken.getContent();
+        if (zenodoToken == null) {
+            LOG.error("Could not get Zenodo token for user " + user.getUsername());
+            throw new CustomWebApplicationException("Could not get Zenodo token for user " +
+                    user.getUsername(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
I would probably also make this a bad request; the use case would be a user tried to create a DOI without having their Zenodo account linked.

Reply:
done